xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/getdocuments.vm

```diff
@@ -99,6 +99,11 @@ $response.setContentType("application/json")
 #set ($limit = $numbertool.toNumber($request.get('limit')).intValue())
 #if (!$limit)
   #set ($limit = 15)
+#else
+  #set ($maxlimit = $services.security.query.getLimit())
+  #if ($isGuest && $limit > $maxlimit)
+    #set ($limit = $maxlimit)
+  #end
```

Comment:
Note that we have a `$mathtool.max()` API.

Reply:
Note that we would normally need a PageTest for this but that's too hard for you to write with your current knowledge. I may add one later on if I find the time.